src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Authorization/AddAuthorizationBuilderFixer.cs

```diff
@@ -0,0 +1,145 @@
+            && arguments[0].Expression is SimpleLambdaExpressionSyntax lambda)
+        {
+            IEnumerable<SyntaxNode> nodes;
+
+            if (lambda.Body is BlockSyntax lambdaBlockBody)
+            {
+                nodes = lambdaBlockBody.DescendantNodes();
+            }
+            else if (lambda.Body is InvocationExpressionSyntax lambdaExpressionBody)
+            {
+                nodes = new[] { lambdaExpressionBody };
+            }
+            else
+            {
+                Debug.Assert(false, "AddAuthorizationBuilderAnalyzer should not have emitted a diagnostic.");
+                return false;
+            }
+
+            var addAuthorizationBuilderMethod = memberAccessExpression.ReplaceToken(identifierToken,
+                SyntaxFactory.Identifier("AddAuthorizationBuilder"));
+
+            invocation = SyntaxFactory.InvocationExpression(addAuthorizationBuilderMethod);
+
+            foreach (var configureAction in nodes)
+            {
+                if (configureAction is InvocationExpressionSyntax { ArgumentList.Arguments: { Count: 2 } configureArguments, Expression: MemberAccessExpressionSyntax { Name.Identifier.Text: "AddPolicy" } })
+                {
+                    invocation = ChainInvocation(
+                        invocation,
+                        "AddPolicy",
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SeparatedList(configureArguments)));
+                }
+                else if (configureAction is AssignmentExpressionSyntax { Left: MemberAccessExpressionSyntax { Name.Identifier.Text: { } assignmentTargetName }, Right: { } assignmentExpression }
+                    && assignmentTargetName is "DefaultPolicy" or "FallbackPolicy" or "InvokeHandlersAfterFailure")
+                {
+                    invocation = ChainInvocation(
+                        invocation,
+                        $"Set{assignmentTargetName}",
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SingletonSeparatedList(
+                                SyntaxFactory.Argument(assignmentExpression))));
+                }
+            }
+
+            return true;
+        }
+
+        Debug.Assert(false, "AddAuthorizationBuilderAnalyzer should not have emitted a diagnostic.");
+        return false;
+    }
+
+    private static InvocationExpressionSyntax ChainInvocation(
+        InvocationExpressionSyntax invocation,
+        string invokedMemberName,
+        ArgumentListSyntax argumentList)
+    {
+        var invocationLeadingTrivia = invocation.GetLeadingTrivia()
+            .Where(trivia => !trivia.IsKind(SyntaxKind.EndOfLineTrivia));
+        var newInvocationTrivia = new SyntaxTriviaList(
+            SyntaxFactory.EndOfLine(
+                Environment.NewLine),
+                SyntaxFactory.Space,
```

Comment:
QQ: is this the only way to create tabs with the SyntaxFactory?

Reply:
I've updated this to use `SyntaxFactory.Tab`, which seems like a better fit.